File: src/main/scala/klaxon/akka/examples/scala/FutureAkka.scala
package klaxon.akka.examples.scala

import java.util.concurrent.TimeUnit

import akka.actor.{ActorLogging, Actor, ActorSystem, Props}
import akka.pattern.ask
import akka.util.Timeout

import scala.concurrent.Await
import scala.concurrent.duration._


/**
 * <p>date 9/17/14 </p>
 * @author klaxon
 */
object FutureAkka {

  implicit val timeout = Timeout(5 seconds)

  def main(args: Array[String]) {
    val actorSystem = ActorSystem("future")
    val fibonacciActor = actorSystem.actorOf(Props[FibonacciActor], "fibonacciActor")

    val futureFibonacci = fibonacciActor ? 12

    println("Working on some time-consuming task. Tick tick.")
    println(s"Calculated fibonacci for 12: " + Await.result(futureFibonacci, 5 seconds))

    actorSystem.shutdown()
  }

}

File: src/main/java/klaxon/akka/examples/java/future/FutureAkka.java
package klaxon.akka.examples.java.future;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.pattern.Patterns;
import scala.concurrent.Await;
import scala.concurrent.Future;
import scala.concurrent.duration.Duration;

import java.util.concurrent.TimeUnit;

/**
 * <p>date 9/17/14 </p>
 *
 * @author klaxon
 */
public class FutureAkka {

    public static void main(String[] args) throws Exception {
        ActorSystem actorSystem = ActorSystem.create("future");
        ActorRef fibonacciActor = actorSystem.actorOf(Props.create(FibonacciActor.class), "fibonacciActor");

        Future<Object> futureFibonacci = Patterns.ask(fibonacciActor, 12, TimeUnit.SECONDS.toMillis(5));

        System.out.println("Working on some time-consuming task. Tick tick.");
        System.out.println("Calculated fibonacci for 12: " + Await.result(futureFibonacci, Duration.create(5, TimeUnit.SECONDS)));

        actorSystem.shutdown();

    }

}
